oeqLinkMigrationLTI/src/main/java/org/apereo/openequella/integration/blackboard/linkmigrationlti/FixerUtils.java

```diff
@@ -421,15 +421,55 @@ private void setNameAndDescription(FixerResponse resp, String textToSearch, int
 	}
   }
 
-  private String removeOldIntegLink(String str, int level, String method) {
+  private String scrubDescription(String str, int level, String method, String[] urls) {
     final String start = "<td><table";
     final String end = "</table></td></tr>";
-     final String toRemove = StringUtils.substringBetween(str, start, end);
-     if(toRemove.contains("href=\"/webapps/dych-tle-")) {
-       return StringUtils.replace(str,start+toRemove+end, "<td/></tr>");
-	 } else {
-       log(level+1, "Not removing old integration link from the body - could not find the 'dych-tle' table");
-       return str;
+     final String[] toRemovePossibilities = StringUtils.substringsBetween(str, start, end);
+     for(String toRemove : toRemovePossibilities) {
+	   if (toRemove.contains("href=\"/webapps/dych-tle-")) {
+		 log(level + 1, "Found old 'dych-tle' table in the body / description.  Replacing old link.");
+		 str = StringUtils.replace(str, start + toRemove + end, "<td/></tr>");
+
+		 // Three possible ways these icons / images are known to appear.
+		 final String[] startImgBlockArr = {"<tr>", "<td", "colspan=\"2\">", "<img", "src=\""};
+		 final String startImgBlock = createIgnoreWhitespaceRegex(startImgBlockArr);
+		 final String[] endImgBlock1Arr = {"\"", "alt=\".\"", "style=", "\"", "border:none;", "width:0px;", "height:4px;", "\"", "/>", "</td>", "</tr>"};
+		 final String endImgBlock1 = createIgnoreWhitespaceRegex(endImgBlock1Arr);
+
+		 final String[] endImgBlock2Arr = {"\"", "alt=\".\"", "style=", "\"", "border:none;", "width:0px;", "height:5px;", "\"", "/>", "</td>", "</tr>"};
+		 final String endImgBlock2 = createIgnoreWhitespaceRegex(endImgBlock2Arr);
+
+		 final String[] startImgBlock3Arr = {"<td", ">", "<img", "src=\""};
+		 final String startImgBlock3 = createIgnoreWhitespaceRegex(startImgBlock3Arr);
+		 final String[] endImgBlock3Arr = {"\"", "alt=\".\"", "style=", "\"", "border:none;", "\"", "/>", "</td>"};
+		 final String endImgBlock3 = createIgnoreWhitespaceRegex(endImgBlock3Arr);
+
+
+		 for (String imgUrl : urls) {
+		   final String wholeImgBlock1 = startImgBlock + imgUrl + endImgBlock1;
+		   final String wholeImgBlock2 = startImgBlock + imgUrl + endImgBlock2;
+		   final String wholeImgBlock3 = startImgBlock3 + imgUrl + endImgBlock3;
+		   log(level + 1, "Removing various forms of the image block from description: " + imgUrl);
+		   str = str.replaceAll(wholeImgBlock1, ""); // remove the <tr ... /> completely
+		   str = str.replaceAll(wholeImgBlock2, ""); // remove the <tr ... /> completely
+		   str = str.replaceAll(wholeImgBlock3, "<td />"); // should be removed when removing the description, but just in case.
+		 }
```

Comment:
Regex parsing HTML feels pretty risky.
There are a lot of quirks and edge cases in the language, would it be feasible to do these transforms using something like jsoup? https://jsoup.org/

Reply:
I can appreciate that.  Using reluctant qualifiers and whitespace wildcards lowers the risk, and we are focused on some very specific use cases for the migration.  I'll see if we can integration `jsoup` without too much hassle.

Reply:
JSoup is now integrated.